src/portals/hope/features/tools/claim-types/tables/TypesTable/BindingForm.tsx

```diff
@@ -0,0 +1,419 @@
+  GetClaimTypeBindingsDocument,
+  GetClaimTypeBindingsQuery,
+  UpdateClaimTypeBindingInput,
+  useCreateClaimTypeBindingMutation,
+  useDeleteClaimTypeBindingMutation,
+  useGetClaimTypeBindingsQuery,
+  useUpdateClaimTypeBindingMutation,
+} from 'types/generated/graphql'
+import {
+  Button,
+  Checkbox,
+  Flex,
+  Input,
+  Keys,
+  useClickOutside,
+  Hint,
+  isPressing,
+  TableColumn,
+  convertCamelcaseToTitle,
+} from '@hedvig-ui'
+import styled from '@emotion/styled'
+import toast from 'react-hot-toast'
+import { ApolloCache, NormalizedCacheObject } from '@apollo/client'
+
+export const TypeColumn = styled(TableColumn)`
+  min-width: 13rem;
+  border-bottom: 1px solid ${({ theme }) => theme.accentContrast} !important;
+  padding: 1rem;
+  position: relative;
+
+  & * {
+    font-size: 12px;
+  }
+
+  &:not(:last-child) {
+    border-right: 1px solid ${({ theme }) => theme.accentContrast};
+  }
+`
+
+const BindingContent = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+`
+
+const BindingTitle = styled.span`
+  color: ${({ theme }) => theme.placeholderColor};
+`
+
+const SetText = styled.span<{ active: boolean }>`
+  color: ${({ theme }) => theme.accent};
+  cursor: pointer;
+  opacity: ${({ active }) => (active ? 1 : 0.3)};
+
+  &:hover {
+    opacity: 0.6;
+  }
+`
+
+const Form = styled.div`
+  position: absolute;
+  z-index: 1000;
+  top: 75%;
+  left: 25%;
+
+  width: 18rem;
+
+  border-radius: 3px;
+  padding: 0.5rem;
+  background-color: ${({ theme }) => theme.backgroundLight};
+
+  & button:hover {
+    background: transparent;
+  }
+`
+
+export const BindingForm: React.FC<{
+  binding?: ClaimTypeBinding | null
+  claimType: string
+  insuranceType: InsuranceType
+}> = ({ binding, claimType, insuranceType }) => {
+  const [createBinding] = useCreateClaimTypeBindingMutation()
+  const [updateBinding] = useUpdateClaimTypeBindingMutation()
+  const [deleteBinding] = useDeleteClaimTypeBindingMutation()
+
+  const { data: bindingData } = useGetClaimTypeBindingsQuery()
+  const typeBindings = bindingData?.claimTypeBindings
+
+  const [showForm, setShowForm] = useState<
+    'defaultReserve' | 'defaultDeductible' | null
+  >(null)
+
+  const [value, setValue] = useState<{
+    defaultReserve: number | null
+    defaultDeductible: number | null
+  }>({
+    defaultReserve: binding?.defaultReserve || null,
+    defaultDeductible: binding?.defaultDeductible || null,
+  })
+
+  useEffect(() => {
+    if (!binding) {
+      return
+    }
+
+    if (
+      binding.defaultDeductible &&
+      binding.defaultDeductible !== value.defaultDeductible
+    ) {
+      setValue((prev) => ({
+        ...prev,
+        defaultDeductible: binding.defaultDeductible || null,
+      }))
+    }
+
+    if (
+      binding.defaultReserve &&
+      binding.defaultReserve !== value.defaultReserve
+    ) {
+      setValue((prev) => ({
+        ...prev,
+        defaultReserve: binding.defaultReserve || null,
```

Comment:
`defaultReserve` would always be there given the `if` above?
```suggestion
        defaultReserve: binding.defaultReserve,
```

Reply:
same with deductible above

Reply:
If I'll remove this one - errors appears. 🤨

<img width="924" alt="image" src="https://user-images.githubusercontent.com/89198006/175471396-36d9644c-69c4-465a-a7f6-a3a1b1dc903d.png">